internal/plans/deferring/deferred.go

```diff
@@ -387,6 +403,24 @@ func (d *Deferred) ReportResourceInstanceDeferred(addr addrs.AbsResourceInstance
 	})
 }
 
+func (d *Deferred) ReportDataSourceInstanceDeferred(addr addrs.AbsResourceInstance) {
```

Comment:
it'd be nice to add a test for this function, even though it's not too complicated